suseconnect-ng.spec

```diff
@@ -121,7 +130,7 @@ find %_builddir/..
 rm -rf %buildroot/usr/share/go
 
 %check
-%gotest -v %import_path/internal/connect
+%gotest -v %import_path/internal/connect -test-hwinfo
```

Comment:
We were talking about running hwinfo here failing on OBS because it is running as a user, but accessing /dev/mem needs root.

There is a way to give builds access to root, but that needs whitelisting by an obs admin, so we should not use it:
https://github.com/openSUSE/open-build-service/blob/af9297d8d8fb79e14951f657a792474c53179e67/src/backend/BSConfig.pm.template#L220

So, no good way to do tests during rpm build that need root. So if that is the only way then it needs to happen in some other CI, like openqa. The openqa tests are currently run against MicroOS on s390, aarch64 and x86_64.

Reply:
ok, so maybe the build service should not run with`-test-hwinfo` and instead do this test as root on openqa instead. Or maybe there is some env var that can be checked to determine if the tests are running openqa, or maybe we could just skip the hwinfo test if `os.Geteuid() != 0`

Reply:
if we add this check, the test will most probably not be run anywhere.
I have another idea: add some macro "flag" in the spec file and make those tests conditional on that. This way we can branch the package, set this flag and get at least the non-root parts executed if needed. These tests have proven useful already and detected missing PPC implementation so I'd rather keep them around at least for manual run.